manim/mobject/geometry.py

```diff
@@ -812,3 +859,233 @@ class RoundedRectangle(Rectangle):
     def __init__(self, **kwargs):
         Rectangle.__init__(self, **kwargs)
         self.round_corners(self.corner_radius)
+
+
+class Tiling(VMobject):
+    """
+    The purpose of this class is to create tilings/tesselations.
+    Tilings can also be seemlessly transformed into each other.
+    This requires their ranges to be the same, tiles to be oriented and
+    rotated the correct way, as well as having a vertex setup that
+    allows proper transformations.
+    
+    Parameters
+    ----------
+    tile_prototype : Mobject or function(x,y) that returns a Mobject
+    x_offset : nested list of Mobject methods and values
+    y_offset : nested list of Mobject methods and values
+    x_range : range
+    y_range : range
+    
+    
+    The tile prototype can be any Mobject (also groups) or a function.
+    The function format is function(x,y), returning a Mobject.
+    Using groups or functions allows the tiling to contain multiple
+    different tiles or to simplify the following offset functions.
+    
+    Next are two nested lists that determine how the tiles are arranged.
+    The functions are typically Mobject.shift and Mobject.rotate.
+    Each list has to contain sublists with Function/Value pairs.
+    More on this in the Examples section.
+
+    Last are two ranges: If both ranges are range(-1,1,1),
+    that would result in a square grid of 9 tiles.
+
+    A Tiling can be directly drawn like a VGroup.
+    Tiling.tile_dictionary[x][y] can be used to access individual tiles,
+    to color them for example.
+    
+    Examples
+    --------
+    The nested lists determining arrangement need more explanation.
+    Example for a shift along the X-Axis, 1 in the positive direction:
+    [[Mobject.shift,[1,0,0]]]
+
+    The origin tile at [x0,y0] won't be moved, but the tile at [x1,y0]
+    will be moved to [1,0,0]. Likewise the tile at [x4,y0] will be moved
+    to [4,0,0]
+    
+    Every step within the tiling applies a full sublist.
+    Example for a shift with simultaneous rotation:
+    [[Mobject.shift,[1,0,0],Mobject.rotate,np.pi]]
+
+    This would move the tile at [x1,y0] to [1,0,0] and rotates it 180°.
+    
+    When multiple sublists are passed, they are applied alternating.
+    Example for alternating shifting and rotating:
+    [[Mobject.shift,[1,0,0]],[Mobject.rotate,np.pi]]
```

Comment:
support highlighting (apply these changes where applicable as well)
```suggestion
    Example for alternating shifting and rotating::
    
        [[Mobject.shift,[1,0,0]],[Mobject.rotate,np.pi]]
```